File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-2.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        if (root == null) {
            return List.of();
        }

        List<List<Integer>> rs = new ArrayList<>();
        Queue<TreeNode> q = new ArrayDeque<>();
        q.offer(root);

        while (!q.isEmpty()) {
            List<Integer> level = new ArrayList<>();
            int qLen = q.size();
            for (int i = 0; i < qLen; i++) {
                TreeNode n = q.poll();
                if (n != null) {
                    level.add(n.val);
                    if (n.left != null) {
                        q.offer(n.left);
                    }
                    if (n.right != null) {
                        q.offer(n.right);
                    }                    
                }
            }
            if (!level.isEmpty()) {
                rs.add(level);
            }            
        }
        
        return rs;
    }
}

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-2.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        self.n = 0
        max = float('-inf')

        def dfs(node, max_val):
            if not node:
                return
            if node.val >= max_val:
                self.n += 1
                max_val = node.val
            dfs(node.left, max_val)
            dfs(node.right, max_val)

        dfs(root, max)
        return self.n


        